quinn-h3/src/client.rs

```diff
@@ -540,6 +544,33 @@ where
     }
 }
 
+#[project]
+impl<B> SendRequest<B, B::Data>
+where
+    B: HttpBody + 'static,
+    B::Error: Into<Box<dyn StdError + Send + Sync>> + Send + Sync,
+{
+    fn take_request(&mut self) -> Result<(Method, Uri, HeaderMap, B), Error> {
+        let (parts, body) = self.request.take().expect("no request").into_parts();
+        let request::Parts {
+            method,
+            uri,
+            headers,
+            ..
+        } = parts;
+
+        match (uri.authority(), headers.get("host")) {
+            (None, None) => return Err(Error::Header("Missing authority")),
+            (Some(a), Some(h)) if a.as_str() != h => {
+                return Err(Error::Header("Host and :authority are in contradiction"));
+            }
+            _ => (),
```

Comment:
You can elide the explicit returns by just having the `Ok((...))` in the `_` case.

Reply:
Much better (valid) style indeed!